### PR TITLE
podman: enable on darwin

### PR DESCRIPTION
**Pull request by marsam**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/podman/disable-go-module-mode.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+diff --git a/Makefile b/Makefile
+index a5265653..1585d04c 100644
+--- a/Makefile
++++ b/Makefile
+@@ -34,10 +34,6 @@ BUILDTAGS ?= \
+ 	varlink
+ 
+ GO_BUILD=$(GO) build
+-# Go module support: set `-mod=vendor` to use the vendored sources
```
Comment (saschagrunert) [inline pkgs/applications/virtualization/podman/disable-go-module-mode.patch]: The `-mod=vendor` should ensure that we do always use the vendored dependencies. So I’m wondering if we really need this patch at all.

Comment (marsam) [reply pkgs/applications/virtualization/podman/disable-go-module-mode.patch]: interesting, I assumed that GO111MODULE=on made network requests for the checksum database, but it does not seem the case here. 
I've removed the patch, thanks!

**Review by saschagrunert — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Pull request merged**